src/diffusers/loaders/lora_pipeline.py

```diff
@@ -2378,6 +2422,14 @@ def _maybe_expand_transformer_param_shape_or_error_(
                     setattr(transformer.config, attribute_name, new_value)
                     logger.info(f"Set the {attribute_name} attribute of the model to {new_value} from {old_value}.")
 
+                    # For `unload_lora_weights()`.
+                    overwritten_params[f"{current_module_name}.weight"] = module_weight
```

Comment:
I think this would have a small but significant memory overhead. For inference purposes only with loras, maybe this could be made opt-out if we know we never want call `unload_lora_weights`. Not a blocker though and can be tackled in a different PR but lmk your thoughts

Reply:
Yeah this could be tackled with `discard_original_layers`.